### PR TITLE
Daml Finance / Update docs for SDK release 2.10.0

### PR DESCRIPTION
**Pull request by johan-da**
Changes related to Daml Finance. This PR:
1. Bumps assembly version (to `1.6.0`) for SDK 2.10.0 and 3.1.
2. Adds the major version `Vx` to path, package, and module names.
3. Adds a script `upgrade_tags.sh` to upgrade references in the docs.
4. Updates changelogs (including list of released and deprecated packages).
5. Adds release notes.
6. Adds links to `AutoCallable` modules (in the package folder).

### CODE REVIEW

**Review by brianweir-da — APPROVED** [review]

**Review by carrielaben-da — APPROVED** [review]

**Review by yves-da — APPROVED** [review]
LGTM, thanks!

But it looks like that few finance doc pages need to be adapted.

File: docs/2.10.0/docs/daml-finance/packages/interfaces/daml-finance-interface-instrument-structuredproduct.rst [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
-###################################################
+Daml.Finance.Interface.Instrument.StructuredProduct.V1
+######################################################
 
 This package contains the *interface* definitions for various structured product instruments. It
 contains the following modules:
 
-- :ref:`BarrierReverseConvertible.Instrument <module-daml-finance-interface-instrument-structuredproduct-barrierreverseconvertible-instrument-73562>`:
+- :ref:`AutoCallable.Instrument <module-daml-finance-interface-instrument-structuredproduct-v1-autocallable-instrument-15420>`:
   Instrument interface for Barrier Reverse Convertibles
```
Comment (markus-da) [inline docs/2.10.0/docs/daml-finance/packages/interfaces/daml-finance-interface-instrument-structuredproduct.rst:11]:
```suggestion
  Instrument interface for Auto Callable
```

File: docs/2.10.0/docs/daml-finance/packages/interfaces/daml-finance-interface-instrument-structuredproduct.rst [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
-- :ref:`BarrierReverseConvertible.Factory <module-daml-finance-interface-instrument-structuredproduct-barrierreverseconvertible-factory-31742>`:
+- :ref:`AutoCallable.Factory <module-daml-finance-interface-instrument-structuredproduct-v1-autocallable-factory-27792>`:
   Factory interface to instantiate Barrier Reverse Convertibles
```
Comment (markus-da) [inline docs/2.10.0/docs/daml-finance/packages/interfaces/daml-finance-interface-instrument-structuredproduct.rst:13]:
```suggestion
  Factory interface to instantiate Auto Callables
```

**Review by markus-da — APPROVED** [review]
Left a few suggestions

File: docs/2.10.0/bin/daml-finance/update_tags.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# Script: update_tags.sh (Daml Finance)
```
[the remark this answers is not in the document]
Comment (yves-da) [reply docs/2.10.0/bin/daml-finance/update_tags.sh:3]: Yes, that's right place. Thanks.

**Pull request merged**